File: atlas-extract-server/Dtos/Enrichment.cs
namespace App.Dtos;



public class Enrichment
{
    public required string ExtractionId { get; set; }
    public required bool GpsEnabled { get; set; }
    public required bool ImagesEnabled { get; set; }
    public string? Country { get; set; }
    public required int TotalItems { get; set; }
    public required int CompletedItems { get; set; }
    public required int FailedItems { get; set; }
    public required string Status { get; set; } // "pending" | "processing" | "completed" | "failed"
    public required DateTime StartedAt { get; set; }
    public DateTime? CompletedAt { get; set; }
    public EnrichmentItem[] Items { get; set; } = [];
}


public class EnrichmentItem
{
    public required string Title { get; set; }
    public required string Status { get; set; } // "pending" | "processing" | "completed" | "failed"
    public string? ErrorMessage { get; set; }
    public string? S3Folder { get; set; } // where the enriched item.json/350.png/1024.png are saved
}

File: atlas-extract-client/src/app/types/Enrichment.ts
export interface EnrichmentItem {
    title: string;
    status: string;
    errorMessage?: string;
    s3Folder?: string;
}

export interface Enrichment {
    extractionId: string;
    gpsEnabled: boolean;
    imagesEnabled: boolean;
    country?: string;
    totalItems: number;
    completedItems: number;
    failedItems: number;
    status: string;
    startedAt: string;
    completedAt?: string;
    items: EnrichmentItem[];
}

export interface EnrichmentStartReq {
    gpsEnabled: boolean;
    imagesEnabled: boolean;
    country?: string;
}

export interface EnrichedItemLocation {
    country?: string;
    state?: string;
    county?: string;
    coordinates?: [number, number];
}

export interface EnrichedItem {
    title: string;
    status: string;
    errorMessage?: string;
    description?: string;
    category?: string;
    tags?: string[];
    location?: EnrichedItemLocation;
    image350Url?: string;
    image1024Url?: string;
}
